tests/mcp/test_server.py

```
@@ -139,6 +139,111 @@ def test_status_all():
+            attempts += 1
+            if attempts == 1:
+                raise RuntimeError("start failed")
+            self.kept = True
+
+    server = KeepGPUServer(
+        controller_factory=cast(Any, lambda **kwargs: FailsOnceController(**kwargs))
+    )
+
+    try:
+        server.start_keep(job_id="retry-job")
+        assert False, "Expected first start to fail"
```

Ruff (0.15.18)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)